common/plugins/com.liferay.ide.core/src/com/liferay/ide/core/project/LiferayNature.java

```diff
@@ -0,0 +1,214 @@
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.core.project;
+
+import com.liferay.ide.core.LiferayCore;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IProjectDescription;
+import org.eclipse.core.resources.IProjectNature;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.wst.jsdt.internal.core.JavaProject;
+
+/**
+ * @author Terry Jia
+ */
+@SuppressWarnings( "restriction" )
+public class LiferayNature implements IProjectNature
+{
+
+    private static final String NATURE_IDS[] = { LiferayCore.NATURE_ID };
+
+    private IProject currentProject;
+    private JavaProject javaProject;
+    private IProgressMonitor monitor;
+    private IPath outputLocation;
+
+    public LiferayNature()
+    {
+        monitor = new NullProgressMonitor();
+    }
+
+    public LiferayNature( IProject project, IProgressMonitor monitor )
+    {
+        currentProject = project;
+
+        if( monitor != null )
+        {
+            this.monitor = monitor;
+        }
+        else
+        {
+            monitor = new NullProgressMonitor();
+        }
+
+    }
+
+    public static void addLiferayNature( IProject project, IProgressMonitor monitor ) throws CoreException
+    {
+        if( monitor != null && monitor.isCanceled() )
+        {
+            throw new OperationCanceledException();
+        }
+
+        if( !LiferayNature.hasNature( project ) )
+        {
+            IProjectDescription description = project.getDescription();
+
+            String[] prevNatures = description.getNatureIds();
+            String[] newNatures = new String[prevNatures.length + LiferayNature.NATURE_IDS.length];
+
+            System.arraycopy( prevNatures, 0, newNatures, 0, prevNatures.length );
+
+            for( int i = 0; i < LiferayNature.NATURE_IDS.length; i++ )
+            {
+                newNatures[prevNatures.length + i] = LiferayNature.NATURE_IDS[i];
+            }
+
+            description.setNatureIds( newNatures );
+            project.setDescription( description, monitor );
+        }
+        else
+        {
+            if( monitor != null )
+            {
+                monitor.worked( 1 );
+            }
+        }
+    }
+
+    @Override
+    public void configure() throws CoreException
+    {
+
+        initOutputPath();
```

Comment:
We don't need the output path or shouldn't need to access the JavProject for now.